### PR TITLE
Poisson EM

### PR DESCRIPTION
**Pull request by grasseau**
New version of the Clustering/Fitting of MCH
This is a draft versIon only to test and compare results with the older algorithm.
A refactored version will be done if the new algorithm works

### CODE REVIEW

**Comment by peressounko** [comment]
It looks like you did not rebased (git pull --rebase) your version before commit?

**Comment by grasseau** [comment]
Ok thank you, I have to do other things before to clean my request

**Review by aphecetche — DISMISSED** [review]
Approve to trigger checks, do not merge yet.

**Review by aphecetche — DISMISSED** [review]

File: Detectors/MUON/MCH/Clustering/include/MCHClustering/dataStructure.h
Comment (ktf) [inline Detectors/MUON/MCH/Clustering/include/MCHClustering/dataStructure.h]: Please make sure you use the guard naming as proposed by the naming convention ( see https://rawgit.com/AliceO2Group/CodingGuidelines/master/coding_guidelines.html?showone=The__define_Guard#The__define_Guard).

**Review by ktf — CHANGES_REQUESTED** [review]
Fix the guards.

File: Detectors/MUON/MCH/Clustering/include/MCHClustering/dataStructure.h
Comment (grasseau) [reply Detectors/MUON/MCH/Clustering/include/MCHClustering/dataStructure.h]: Thank you for this mistake and the documentation.
A new re-factorized version is in preparation. This PR is only to test numerically 
the algorithm (the original clustering is unmodified).

**Review by aphecetche — DISMISSED** [review]

**Review by aphecetche — APPROVED** [review]

**Comment by aphecetche** [comment]
@TimoWilken @ktf would need some help here to understand what's wrong in the fullCI build : the summary page is kind of empty and I'm at lost in the full log to find the issue...

**Comment by davidrohr** [comment]
sim_challenge returns non 0 exit code:
```
++ grep 'Return status' sim-challenge.log
++ grep -v ': 0'
Return status of Efficiency: 1
```

For reference, Sandro has just improved this yesterday night, so in the future it should print reasonable errors.

**Comment by aphecetche** [comment]
@ktf if I'm not wrong the include guards have been changed. Could you clear/validate your change request if that's so ?

**Comment by ktf** [comment]
Apart from the fact that almost everywhere else it is O2 and not ALICEO2, I guess the guards are fine.

**Comment by aphecetche** [comment]
> Apart from the fact that almost everywhere else it is O2 and not ALICEO2, I guess the guards are fine.

well, this is something I can easily prepare another PR for (in `MUON/*` directories)

**Comment by aphecetche** [comment]
@ktf could you clear your change request so I can merge this one (well, it's no longer green for the moment due to some failure in fullCI, but it was green yesterday, so I assume it will be back to green at some point...)

**Comment by ktf** [comment]
AFAICT, it it clean, no?

**Comment by aphecetche** [comment]
@ktf well, I still see the "Changes requested" banner before the checks and then `"Merging is blocked" Merging can be performed automatically once the requested changes are addressed` after the checks.

**Review by ktf — APPROVED** [review]

**Comment by ktf** [comment]
That's because the fullCI test is failing actually. Given it is unrelated, I am merging this PR.

**Pull request merged**
